src/App.jsx

```diff
@@ -14,33 +20,100 @@ export const goodsFromServer = [
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getPrepearedGoods(goods, { sortField, isReversed }) {
+  const prepearedSortGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortField) {
+    prepearedSortGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed === true) {
+    prepearedSortGoods.reverse();
+  }
+
+  return prepearedSortGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const sortGoods = getPrepearedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
+  const reset = () => {
+    setSortField(SORT_FIELD_RESET);
+    setIsReversed(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const isOriginalOrder =
+    JSON.stringify(sortGoods) === JSON.stringify(goodsFromServer);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_FIELD_ALPHABET,
+          })}
+          onClick={() => setSortField(SORT_FIELD_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SORT_FIELD_LENGTH,
+          })}
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(prev => !prev)}
+        >
+          Reverse
+        </button>
+
+        {!isOriginalOrder && (
+          <button
+            type="button"
+            className={cn('button', 'is-danger', {
+              'is-light': sortField !== SORT_FIELD_RESET,
+            })}
+            onClick={reset}
+          >
+            Reset
+          </button>
+        )}
+      </div>
+
+      <ul>
+        {sortGoods.map(good => (
+          <li data-cy="Good" key={good.id}>
```

Comment:
You haven't id here 
<img width="500" alt="image" src="https://github.com/mate-academy/react_list-of-goods-js/assets/64750958/f930fe99-1b5f-4f77-9588-3e3396e285f7">
```suggestion
          <li data-cy="Good" key={good)>
```